src/constructor/load.py

```diff
@@ -0,0 +1,221 @@
+import logging
+import torch
+import pytorch_lightning as pl
+from pytorch_lightning.utilities.cloud_io import load
+from typing import Optional, Dict, List, Callable, OrderedDict, Union
+
+
+def load_state_dict(checkpoint_path: str, map_location: Optional[Union[str, Callable, torch.device]] = 'cpu'):
+    """Loads a checkpoint state_dict.
+
+    Args:
+        checkpoint_path: Path or URL of the checkpoint, it can be S3 on AWS, GCS on Google Cloud, or ADL on Azure.
+        map_location: How to remap storage locations.
+
+    Returns:
+        state_dict: Downloaded state dict.
+    """
+    checkpoint = load(checkpoint_path, map_location=map_location)
+    if 'state_dict' in checkpoint:
+        state_dict = checkpoint['state_dict']
+    else:
+        state_dict = checkpoint
+    return state_dict
+
+
+def check_state_dict_keys(checked_state_dict: OrderedDict[str, torch.Tensor], model_keys: List[str], 
```

Comment:
Yes we really need this. Let's explain me the case.
User put the base checkpoint with all keys, then he put overridden checkpoint with for example only half the weights.
So in this case `nn.Module.load` will not raise the error because missing half of overridden checkpoint would be get from base checkpoint. 

In fact it depend on implementation of load method, now it update the dict, but if we remove from dict all the straightened keys from overridden checkpoint it will raise by nn.Module.load.

But the next thing won't let me do it. More code in generating state dict, raise will not send you information in which checkpoint arose the error.